File: cpp/Mod_nextSibling.cpp
Node nextSibling(Tree t, Node v) {
    Node res = createNode(-1);
    int cnt = 0;
    Node curr;
    List<Node> ln = children(t, parent(t,v));
    
    while (length(ln) == 1) {
        curr parent(t, v);
        cnt++;
    }
    
    return res;
}

File: hs/Mod_nextSibling.hs
module Mod_nextSibling ( nextSibling ) where
import Prelude hiding (and, or) 
import Debug.Trace
import Splib
import Types

nextSibling tree node = get (children tree (parent tree node)) ((findIndex tree node 0)+1)

findIndex tree node i | get (childs) i == node = i
                      | otherwise = findIndex tree node (i+1)
                    where
                     childs = children tree (parent tree node)